src/NHibernate/Criterion/Projections.cs

```diff
@@ -356,6 +356,18 @@ public static IProjection Conditional(ICriterion criterion, IProjection whenTrue
 			return new ConditionalProjection(criterion, whenTrue, whenFalse);
 		}
 
+		/// <summary>
+		/// Conditionally returns one of the <see cref="ConditionalProjectionCase.Projection"/>s depending on the <see cref="ConditionalProjectionCase.Criterion"/>s of <paramref name="cases"/> or the <paramref name="elseProjection"/>.
+		/// This produces an switch-case expression with multiple when-then parts.
+		/// </summary>
+		/// <param name="cases">The <see cref="ConditionalProjectionCase"/>s which contain your <see cref="ICriterion"/>s and <see cref="IProjection"/>s.</param>
+		/// <param name="elseProjection">The else <see cref="IProjection"/>.</param>
+		/// <returns>A <see cref="IProjection"/> for a switch-expression with multiple <see cref="ICriterion">Criterions</see> ("when") <see cref="IProjection">Projections</see> ("then").</returns>
+		public static IProjection Conditional(ConditionalProjectionCase[] cases, IProjection elseProjection)
```

Comment:
Maybe better make it `IList<ConditionalProjectionCase> cases`?

Reply:
Hi, yes i would agree but it should be an array due to performance. You could pass an array to IList (the interface doesn't really change the performance), but that would move the control of the performance to the caller: they need to "know to pass an array" if they care about performance (and we do, we pass 100+ cases multiple times a second).
Yeah it's not much, but if we have the choice?

Reply:
I don't get your point about `know to pass an array`. Why  `List<ConditionalProjectionCase>` or any other collection that implements `IList<>` would perform worse than array?

Reply:
I'll just leave this as is.

Reply:
> 
> 
> I don't get your point about `know to pass an array`. Why `List<ConditionalProjectionCase>` or any other collection that implements `IList<>` would perform worse than array?

Just to conclude my point: :) 
Using IList<> enables you to deliver every better or worser IList<>-type like "slower" List<> instead of an array.
The current implemention forces you to deliver an array - which is much faster to iterate than a - probably - used List<>.
A developer may not know that the implementation's performance relies on the use of thhe delivered real type (List<>, Array[], ...) for the IList<>.
And as we don't change the array, we don't need IList<> which only enables you to add something (what we don't do).
That's my reason to force everyone to use array.
We may discuss whether to use IReadOnlyCollection https://docs.microsoft.com/en-us/dotnet/api/system.array.asreadonly?view=net-5.0 
but that enables you to use an List<> again - which is a bit slower. It's not much difference, but we take every bit performance we can get.
Or we create a second cctor with IList<> as "a backdoor" for using other types - if its really that neccessary - and write a note to documentation "performance depends on real used type" or something like that.

Have a nice day and thank you all for accepting :) 
Paul